server/build_event_protocol/build_event_handler/build_event_handler.go

```diff
@@ -814,6 +814,13 @@ func (e *EventChannel) FinalizeInvocation(iid string) error {
 	if err != nil {
 		return err
 	}
+	auth := e.env.GetAuthenticator()
+	userInfo, err := auth.AuthenticatedUser(e.ctx)
+	groupID := ""
+	if err == nil {
+		groupID = userInfo.GetGroupID()
+	}
+	ti.GroupID = groupID
```

Comment:
are there going to be edge cases where the group id is not set in the context? maybe it's safer to record the group id in the metrics but not updating the field in the db.

Reply:
this feels like a little bit of a weird place to be doing this -- I'd expect we have groupID already in hand when we validate the context at the start of the invocation. Maybe we just need to hold on to it in the channel or something?

Reply:
I implemented a function getGroupIDForMetrics similar to what we did in execution server ([code](https://github.com/buildbuddy-io/buildbuddy/blob/83d8cbe404d069d193236dedf35d6654dc430973/enterprise/server/remote_execution/execution_server/execution_server.go#L743)). So that we are not updating the database.

We validated the group id inside `CreateInvocation` method. So we need to refactor it in order to put it in the channel.